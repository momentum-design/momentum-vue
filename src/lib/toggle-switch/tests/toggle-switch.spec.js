import { shallowMount } from '@vue/test-utils';
import ToggleSwitch from '../index.vue';

describe('ToggleSwitch', () => {
  it('should match snapshot', () => {
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123'
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one ToggleSwitch', () => {
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123'
      }
    });

    expect(container.findAll('input').length).toEqual(1);
  });

  it('should render label with two child spans', () => {
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123'
      }
    });

    expect(container.findAll('label').length).toEqual(1);
    expect(container.find('label').findAll('span').length).toEqual(2);
  });


  it('should handle disabled state', () => {
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123',
        disabled: true
      }
    });

    expect(container.find('input').attributes().disabled).toEqual('disabled');
  });

  it('should handle onChange event', () => {
    let count = 0;
    const countUp = () => count++;
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123'
      },
      listeners: {
        change: countUp
      }
    });

    container.find('input').trigger('change');
    expect(count).toEqual(1);
  });

  it('should handle onClick event', () => {
    let count = 0;
    const countUp = () => count++;
    const container = shallowMount(ToggleSwitch, {
      propsData: {
        label: 'test',
        htmlId: 'test123'
      },
      listeners: {
        change: countUp
      }
    });

    container.find('input').trigger('click');
    expect(count).toEqual(1);
  });
});

