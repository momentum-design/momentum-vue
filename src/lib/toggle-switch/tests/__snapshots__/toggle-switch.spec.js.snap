// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch should match snapshot 1`] = `<div class="md-input-container md-toggle-switch"><input type="checkbox" name="" id="test123" tabindex="0" class="md-input md-toggle-switch__input" value=""> <label for="test123" class="md-toggle-switch__label"><span class="md-toggle-switch__label__container"></span> <span class="md-toggle-switch__label__text">test</span></label></div>`;
